README.md

```diff
@@ -47,28 +83,121 @@ Thats it!!
-| `fail-when-not-valid` | Fails the job if the branch is not valid.                   | no |     true      |
+## **Action Outputs**
+</div>
+
+| Input Name            | Description                                                 | Values Returned  |
+|-----------------------|:------------------------------------------------------------|:----------------:|
+| `branch-name`         | Returns a string value of 'true' or 'false' indicating whether or not the branch is valid.  | 'true' or 'false' |
 
 ---
 
 <div align="center" style="font-weight:bold">
 
-## **Examples**
+## **Validation Logic Expression Functions**
 </div>
 
+Below is a small list of the available expression functions that you can use in the value of the `validation-logic` input.  These expression functions can be used in combination using the `&&` and `||` operators.  
+  - Example: equalTo('feature/my-*-branch') && allLowerCase()
+    - This would check to see if the branch is equal to the value and that the entire branch is lower case.
+
 <div align="center">
 
-### **Fail the job if the package is not found**
+### **Expression Function List**
+</div>
+
+1. `equalTo(string)` - Checks to see if the branch name given is equal to the argument value of the function.  The argument value must be a string value that is surrounded by single or double quotes.  The quotes used must be the opposite of the quotes used for the entire input value.  Standard YAML syntax rules apply.
+    The function value allows the use of 2 characters that provide globbing like behavior.  The 2 characters are `#` and `*` and can be used together as many times as needed.
+     - Example 1: equalTo('main')
+       - This will check if the branch is equal to the value of `main`.
+     - Example 2: equalTo('feature/my-*-branch')
+       - This will check if the branch starts with the value `feature/my-` and that it ends with `-branch`.  Anything is acceptable between the beginning and end of the branch where the `*` character is located.  This should be a familiar concept to other systems and uses of this type of syntax.
+     - Example 3: equalTo('feature/#-sample-branch')
+       - This would return valid if the branch name was `feature/my-sample-branch`.  This will check if the branch starts with the value `feature/` and ends with the value `-sample-branch`.  Any text between the start and the end will be checked to see if it is a whole number of any digit size.
+         - This would return as valid if the name of the branch was `feature/12-sample-branch` or `feature/12345-sample-branch`.
+         - This would return as not valid if the name of the branch was `feature/10-20-sample-branch`.  This is because in this example, the branch ends with the value `-20-sample-branch`, not `-sample-branch`.
+     - Example: 4: equalTo('release/v#.#.#-*.#')
+       - The would return valid if the branch name was `release/v1.2.3-preview.4`.
+2. `allLowerCase()` - Check if the branch name is all lower case
```

Comment:
Refactor "if" to "whether or not"  (where it makes sense) and make sure to look at the verb.  In this case, it will be "Checks" not "Check"

```suggestion
2. `allLowerCase()` - Checks whether or not the branch name is all lower case
```